gsoc2018-bharat/src/evaluate.py

```diff
@@ -0,0 +1,205 @@
+        r = np.array(list(map(lambda x: wv.get_vector(x), d)),
+                     dtype=np.float32)
+        hidden = model.init_hidden()
+    except KeyError:
+        return np.random.randn(100)
+    try:
+        for word in r:
+            p, hidden = model(Variable(torch.tensor([[word]])),
+                              hidden)
+            p = p[0][0].detach().numpy()
+        return p
+    except (KeyError, IndexError, TypeError) as _:
+        return np.random.randn(100)
+
+
+def evaluate():
+    global dictionary, wv
+    count = 0
+    scores = np.zeros(7)
+    cos_scores = np.zeros(7)
+    itr = len(dictionary)
+    logging.info('running evaluation for {0} samples'.format(itr))
+    for key in dictionary:
+        progress = (count / itr) * 100
+        print('INFO : evaluated {0:.1f} %'.format(progress), end='\r')
+        d = dictionary[key].split('resource/')
+        d = [idx.split()[0].translate(table).lower() for idx in d[1:]]
+        try:
+            r = np.array(list(map(lambda x: wv.get_vector(x), d)),
+                         dtype=np.float32)
+        except KeyError:
+            itr -= 1
+            continue
+        if np.any(np.isnan(r)):
+            itr -= 1
+            continue
+        else:
+            if r.ndim == 2:
+                try:
+                    r = r.mean(axis=0).reshape(1, -1)
+                    vec1 = mean_encoder(dictionary[key]).reshape(1, -1)
+                    vec2 = distance_encoder(dictionary[key]).reshape(1, -1)
+                    vec3 = title_mean(key).reshape(1, -1)
+                    vec4 = abstract_encoder(key).reshape(1, -1)
+                    vec5 = np.random.randn(100).reshape(1, -1)
+                    vec6 = np.zeros(100).reshape(1, -1)
+                    t = wv.similar_by_vector(vec4)
+                    scores[0] += norm(r - vec1)
+                    scores[1] += norm(r - vec2)
+                    scores[2] += norm(r - vec3)
+                    scores[3] += norm(r - vec4)
+                    scores[4] += norm(r - vec5)
+                    scores[5] += norm(r - vec6)
+                    scores[6] += wv.similar_by_vector(vec4)[0][1]
+                    cos_scores[0] += cosine_similarity(r, vec1)
+                    cos_scores[1] += cosine_similarity(r, vec2)
+                    cos_scores[2] += cosine_similarity(r, vec3)
+                    cos_scores[3] += cosine_similarity(r, vec4)
+                    cos_scores[4] += cosine_similarity(r, vec5)
+                    cos_scores[5] += cosine_similarity(r, vec6)
+                    cos_scores[6] += wv.similar_by_vector(vec4)[0][1]
+                    count += 1
+                except (ValueError, KeyError) as _:
+                    itr -= 1
+                    continue
+            else:
+                itr -= 1
+                continue
+    scores = scores / norm(scores)
+    cos_scores = cos_scores / norm(cos_scores)
+    print(scores)
+    print(cos_scores)
```

Comment:
maybe the some data strucutre should be returned, and then you can code up a `print_summary` function that displays that information somehow.